polkadot/node/core/pvf/common/src/executor_intf.rs

```diff
@@ -140,21 +140,20 @@ pub unsafe fn create_runtime_from_artifact_bytes(
 	executor_params: &ExecutorParams,
 ) -> Result<WasmtimeRuntime, WasmError> {
 	let mut config = DEFAULT_CONFIG.clone();
-	config.semantics =
-		params_to_wasmtime_semantics(executor_params).map_err(|err| WasmError::Other(err))?;
+	config.semantics = params_to_wasmtime_semantics(executor_params);
 
 	sc_executor_wasmtime::create_runtime_from_artifact_bytes::<HostFunctions>(
 		compiled_artifact_blob,
 		config,
 	)
 }
 
-pub fn params_to_wasmtime_semantics(par: &ExecutorParams) -> Result<Semantics, String> {
+pub fn params_to_wasmtime_semantics(par: &ExecutorParams) -> Semantics {
 	let mut sem = DEFAULT_CONFIG.semantics.clone();
 	let mut stack_limit = if let Some(stack_limit) = sem.deterministic_stack_limit.clone() {
 		stack_limit
 	} else {
-		return Err("No default stack limit set".to_owned())
+		panic!("No default stack limit set");
```

Comment:
Just to double-check here, this is following up on a discussion with @s0me0ne-unkn0wn right? Can you remind me what the reasoning was?

Reply:
Yep. Coz that would be triggered when the default config is messed up. Deliberately or not, it's a sign of suicidal behavior, which should never happen. That's what you told me lol.

Reply:
I see we have a comment on `DEFAULT_CONFIG` saying never to change it. So I think instead of a `panic!` we can make this an `expect` instead, with a [proof](https://github.com/paritytech/polkadot-sdk/blob/master/docs/STYLE_GUIDE.md#style). Or, since that doc says to avoid panics where possible, we can make some const like `DEFAULT_STACK_LIMIT` and refer to it and have `DEFAULT_CONFIG` also refer to it. Though that might require `clone` which I'm not sure is possible in a const context. Maybe `DeterministicStackLimit` can be `Copy` though.

Reply:
> So I think instead of a panic! we can make this an expect instead, with a [proof](https://github.com/paritytech/polkadot-sdk/blob/master/docs/STYLE_GUIDE.md#style).

But that's just a subtler way of doing what we do right now.

```rs
let mut stack_limit = sem.deterministic_stack_limit.clone()
	.expect("Default stack limit should always be available");

// louder, noisier, but maybe this is exactly what we want?
let mut stack_limit = if let Some(stack_limit) = sem.deterministic_stack_limit.clone() {
	stack_limit
} else {
	panic!("No default stack limit set");
};
```

The only way this could panick is when someone changes the code, which is not supposed happen ever I guess?

Reply:
I don't see why we'd reimplement `expect` here TBH. 

> The only way this could panick is when someone changes the code, which is not supposed happen ever I guess?

Exactly, we can write a proof stating this, and that there is a comment saying not to change the code.

Reply:
As I said, `panic!` is louder and stands out in the code. But I didn't write the code. That's just my guess. :))

Reply:
I don't think it needs to be loud here; we can have a proof, and there is already a loud comment on the config. Sorry if we gave a different impression, I honestly don't remember the original conversation.